ledger/src/blockstore.rs

```diff
@@ -4616,6 +4616,39 @@ pub mod tests {
         assert_eq!(result, data);
     }
 
+    #[test]
+    fn test_multi_get() {
+        const TEST_PUT_ENTRY_COUNT: usize = 100;
+        let ledger_path = get_tmp_ledger_path_auto_delete!();
+        let blockstore = Blockstore::open(ledger_path.path()).unwrap();
+
+        // Test meta column family
+        for i in 0..TEST_PUT_ENTRY_COUNT {
+            let k = u64::try_from(i).unwrap();
+            let meta = SlotMeta::new(k, Some(k + 1));
+            blockstore.meta_cf.put(k, &meta).unwrap();
+            let result = blockstore
+                .meta_cf
+                .get(k)
+                .unwrap()
+                .expect("Expected meta object to exist");
+            assert_eq!(result, meta);
+        }
+        let mut keys: Vec<u64> = vec![0; TEST_PUT_ENTRY_COUNT];
+        for (i, key) in keys.iter_mut().enumerate().take(TEST_PUT_ENTRY_COUNT) {
+            *key = u64::try_from(i).unwrap();
+        }
+        let values = blockstore.meta_cf.multi_get(keys);
+        for (i, value) in values.iter().enumerate().take(TEST_PUT_ENTRY_COUNT) {
+            let k = u64::try_from(i).unwrap();
+            if let Ok(Some(v)) = value {
+                assert_eq!(v, &SlotMeta::new(k, Some(k + 1)));
+            } else {
+                panic!();
+            }
```

Comment:
Full disclosure, I didn't actually try to compile above code before posting and I now know it didn't compile as-is 😄 :
- The outer for-loop is a iterating over a `Vec<Result<Option<SlotMeta>>>`
- So, within the loop, value is a `&Result<std::option::Option<SlotMeta>>`
- We need to get down to the `SlotMeta` for the comparison; `as_ref().unwrap()` peels off the Result to yield `&Option<SlotMeta>`
- Another `as_ref().unwrap()` peels off the Option to yield `&SlotMeta`

Altogether, a line that would actually compile is:
```rust
assert_eq!(value.as_ref().unwrap().as_ref().unwrap(), &SlotMeta::new(k, Some(k + 1)));
```
The `as_ref()`'s are necessary because we used `.iter()` and could only borrow from the `Vec<_>`. Using `into_iter()` in the for-loop would consume the Vec and return items instead of references. In this case, the below line would compile (noting no borrow now on the rhs SlotMeta):
```rust
assert_eq!(value.unwrap().unwrap(), SlotMeta::new(k, Some(k + 1)));
```
Either of these approaches are fine with me